### PR TITLE
Fixer.io provider

### PR DESCRIPTION
**Pull request by harpener**
https://business-factory.github.io/docs/tasks/python/fixer-io-currency-provider

@business-factory/pythonistas PRM.

### CODE REVIEW

**Comment by harpener** [comment]
@business-factory/pythonistas I refactored the provider a bit, please take a look :-)

**Comment by miso-belica** [comment]
I can see that response from https://api.fixer.io/2017-09-22?base=USD looks like this
```json
{
    "base": "USD",
    "date": "2017-09-21",
    "rates": {
        "AUD": 1.2601,
        "...": 1.6428,
    }
}
```

That means that this API may return rates for another day `"date": "2017-09-21"` as requested `2017-09-22`. @martinvy @harpener Do you think it's OK and it will be overwritten next day or we should somehow check it?

File: gold_digger/data_providers/fixer.py [hunk 1]
```diff
@@ -25,13 +25,20 @@ def get_all_by_date(self, date_of_exchange, currencies):
...
 
-        for currency in currencies:
-            decimal_value = self._get_by_date(date_of_exchange_string, currency)
-            if decimal_value is not None:
-                day_rates[currency] = decimal_value
+        url = self.BASE_URL.format(date=date_of_exchange_string)
+        response = self._get(url, params={"base": self.BASE_CURRENCY})
+
+        if response:
+            response = response.json()
```
Comment (harpener) [inline gold_digger/data_providers/fixer.py]: @martinvy It isn't wrapped in try-except block here, but I guess it should not fail anyway.

Comment (martinvy) [reply gold_digger/data_providers/fixer.py]: You are right. But I can fix it.

**Comment by miso-belica** [comment]
@martinvy @harpener  Yes, when I run the command with clean DB everything is fine, but when I run the command second time I get a lot of errors like this:

```
[WARNING] 2017-09-26 08:54:12 at dao_exchange_rate.py:26 (MainProcess) -- Session rollback. Error: (psycopg2.IntegrityError) duplicate key value violates unique constraint "USD_exchange_rates_date_provider_id_currency_key"
DETAIL:  Key (date, provider_id, currency)=(2017-09-24, 2, HKD) already exists.
[SQL: 'INSERT INTO "USD_exchange_rates" (date, provider_id, currency, rate) VALUES (%(date)s, %(provider_id)s, %(currency)s, %(rate)s) RETURNING "USD_exchange_rates".id'] [parameters: {'rate': Decimal('7.8121039999999997149870978319086134433746337890625'), 'currency': 'HKD', 'provider_id': 2, 'date': datetime.date(2017, 9, 24)}]. Record {'provider_id': 2, 'rate': Decimal('7.8121039999999997149870978319086134433746337890625'), 'date': datetime.date(2017, 9, 24), 'currency': 'HKD'}
```

**Comment by martinvy** [comment]
@miso-belica well, that's not quite an issue, but the UX could be better :) The error refers to inserting of duplicated exchange rate, but the duplicates are only logged and then skipped and all other rates are correctly inserted.

**Pull request merged**